templates/loginform.php

```diff
@@ -280,13 +281,23 @@ class="float-l erroricon" style="margin: 15px "/>
     SimpleSAML_Logger::debug("signRequests: " . print_r($signRequests, TRUE));
     SimpleSAML_Logger::debug("signRequests json: " . json_encode($signRequests, TRUE));
     echo '<script type="text/javascript">';
-	echo 'var signRequests = ' . json_encode($signRequests) . ';';
-	echo '    u2f.sign(signRequests, function (result) {
-                console.log(result);
-                document.getElementById("signatureData").value = result.signatureData;
-                document.getElementById("clientData").value = result.clientData;
-                document.forms["piLoginForm"].submit();
-        });';
+    if (isset($this->data['serial'])) {
```

Comment:
Is there another way than the availability of the serial number to determin if this is the second step of the registration process?

Reply:
Please have a look at this d577235